### ISSUE TITLE
Multiple dynamic content blocks show multiple editors

### ISSUE THREAD

**Issue by mallezie** [mautic/mautic#14288]
### Mautic Series

5.1.x series

### Mautic installed version

5.1.1

### Way of installing

I installed with composer using https://github.com/mautic/recommended-project

### PHP version

8.1

### What happened?

When i create an email with blank template and edit it with the grapesjs builder i'm seeing the editor in a dynamic content block multiple times when i use multiple dynamic content blocks. 
![image](https://github.com/user-attachments/assets/32cd2bee-54b2-49f1-8963-fb289ae592f1)

### How can we reproduce this issue?

1.  Create new email select blank theme (problem also occurs in other themes)
2. Add dynamic content block, no need to create variants
3. Add a second dynamic content block, no need to create variants. 
4. Close and reopen the second dynamic content block, and see the editor showing up multiple times. 

Optional: Create more blocks or variants in the first 2 blocks, and see the number of editors shown grow. 
![image](https://github.com/user-attachments/assets/6224b4c7-7960-41a5-ba25-7e4a9765cb3f)

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Thanks for the detailed bug report @mallezie - really helpful!

**Comment by mallezie** [comment]
I've investigated the issue of multiple CKEditors appearing within dynamic content blocks and believe I've pinpointed the source of the problem. However, I'm no JS specialist in this area, so not sure how to solve it or pinpoint it further. 
Just posting my findings here. 

**Findings:**

* **Triggering Events:**
    * When a new tab (variation) is added to a dynamic content block (using "add variation"), or when a new dynamic content block is created, the `grapesjs-preset-mautic` node module (found here: [https://github.com/mautic/grapesjs-preset-mautic](https://github.com/mautic/grapesjs-preset-mautic)) executes the following:

    ```javascript
    // When a new Dynamic Content filter (tab) is added, we want to turn the editor into CKEditor.
    Mautic.dynamicContentAddNewFilterListener((textarea) => {
      Mautic.ConvertFieldToCkeditor(textarea, {});
    });

    // When a new Dynamic Content item (slot) is added, we want to turn the editor into CKEditor.
    Mautic.dynamicContentAddNewItemListener((textarea) => {
      Mautic.ConvertFieldToCkeditor(textarea, {});
    });
    ```

    * This code essentially attaches listeners to the textarea, instructing it to be converted into a CKEditor instance.

* **Listener Implementation:**
    * Within the `emailBundle` JavaScript code (specifically [https://github.com/mautic/mautic/blob/5.2.3/app/bundles/EmailBundle/Assets/js/email.js#L382](https://github.com/mautic/mautic/blob/5.2.3/app/bundles/EmailBundle/Assets/js/email.js#L382)), the listener registration is implemented as follows:

    ```javascript
    Mautic.dynamicContentAddNewItemListener = function(callback) {
        Mautic.internalDynamicContentItemCreateListeners.push(callback);
    }
    ```

    * And a similar pattern is used for `dynamicContentAddNewFilterListener`.

* **Listener Execution:**
    * The listeners are then executed here: [https://github.com/mautic/mautic/blob/5.2.3/app/bundles/EmailBundle/Assets/js/email.js#L364](https://github.com/mautic/mautic/blob/5.2.3/app/bundles/EmailBundle/Assets/js/email.js#L364)

    ```javascript
    if (Mautic.internalDynamicContentItemCreateListeners) {
        Mautic.internalDynamicContentItemCreateListeners.forEach(function(callback) {
            callback(textarea);
        });
    }
    ```

    * This code iterates through the registered listeners and executes the `callback` function, which triggers the CKEditor conversion.

* **Problem Identification:**
    * The core issue appears to be that the listeners are being triggered and executed multiple times for a single textarea element. This results in multiple CKEditor instances being created on top of each other.

**Comment by LordRembo** [comment]
@mallezie I believe I found a fix. It's added to the `grapesjs-preset-mautic` submodule of the GrapesJS plugin:
https://github.com/mautic/grapesjs-preset-mautic/pull/56

This does mean that, in order to get things fixed, you need to test & validate that merge request.

**To test:**

1. Have you Mautic test project ready
2. inside the GrapesJS Plugin (`plugins/GrapesJsBuilderBundle`), replace the `grapesjs-preset-mautic` node module with the version [from this merge request](https://github.com/LordRembo/grapesjs-preset-mautic/tree/fix/dynamic-content-multiple-editors): `npm i grapesjs-preset-mautic@github:LordRembo/grapesjs-preset-mautic/tree/fix/dynamic-content-multiple-editors`
3. Rebuild the GrapesJS Plugin javascript using `npm run rebuild` (still in `plugins/GrapesJsBuilderBundle`)

**Validate**

Once validated, we can merge the preset changes here: https://github.com/mautic/grapesjs-preset-mautic/pull/56
And make a PR for Mautic to include the changed module.

**Comment by abhisekmazumdar** [comment]
I can confirm that the mentioned MR: https://github.com/mautic/grapesjs-preset-mautic/pull/56
fixes this issue i.e. I don't see double inputs variation box of dynamic content blocks when adding multiple dynamic content blocks

**Comment by LordRembo** [comment]
I added a New PR to integrate the fix, if anyone can test it: https://github.com/mautic/mautic/pull/14925
